### ISSUE TITLE
Soooo --all-namespaces should be the default

### ISSUE THREAD

**Issue by ORESoftware** [kubernetes/kubernetes#79678]
So I started using namespaces for the first time, and I ran:

```bash
kubectl get pods
```

nothing was showing up...I was really confused, debugging stuff for 30+ minutes..
finally I was like, even tho this is really dumb, maybe they are behind a namespace, so I did:

```bash
kubectl get pods --namespace=mine
```

and finally my pods showed up.  so yes there is:

```bash
kubectl get pods --all-namespaces
```

but this should really be the default, shouldn't it?  You will save people a lot of time if you make this fix. tbh, this is a really crappy cli design in this case.

**Comment by almiche** [comment]
While this might sound like a good idea if you're running a single deployment with a few pods, once you start running multiple dozens of deployments and objects it starts to become less and less useful to have this become the default. I also suspect exposing all objects without specifying namespace is probably a non-goal. Namespaces were created for a reason which is to logically segment resources together. In my opinion, what you are suggesting just nullifies the entire purpose of namespaces.

**Comment by ramnar** [comment]
I think this is more of a usability bug. 

**kubectl get pods**  gives the list of all pods in default namespace.By default no pods are present in default namespace. So it does give a message saying **No Resources found**. 

Improving this message a little bit will be helpful like **No Resources found in default namespace**

**Comment by ramnar** [comment]
@kubernetes/sig-cli
/sig cli

**Comment by almiche** [comment]
@ramnar I could implement this if that's alright with you !

**Comment by ORESoftware** [comment]
sure, how about 

```
No Resources found in default namespace. Use --namespace=x to search in other namespaces.
```

**Comment by ramnar** [comment]
> @ramnar I could implement this if that's alright with you !

We should wait for kubernetes members to comment if this change should be taken up

**Comment by oscarlofwenhamn** [comment]
The title of this issue could use an update, if the priority has been shifted to improving the output of an empty `default` namespace. Maybe it's early to claim the consensus seems to be that --all-namespaces should **not** be the default, but I personally agree that it shouldn't be. 

Take note of the shorthand `-A`, to make typing it a bit easier, or the option to [set the namespace preference](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/#setting-the-namespace-preference) to something other than `default` to get around typing your namespace altogether.

**Comment by nagexiucai** [comment]
@ORESoftware @almiche @ramnar @oscarlofwenhamn :)

IMHO, `default` is innocent.
But we could make the messages more clear.
For example, `kubectl get xxx -n nonexistent-namespace` will show `No Resources found` also.

And `--all-namespaces` is a bit long, but a `-a/-A/--all/--ALL` is too simple to describe our intent, "all namespaces of cluster or all some resource of some namespace", so puzzled.

My conclusion: `--all-namespaces` is competent, `default` namespace is ok, making more clear message is enough.

**Comment by oscarlofwenhamn** [comment]
@nagexiucai "`-A`~`--all-namespaces`" is already implemented as in #72006, in [v1.14.0-alpha.1](https://github.com/kubernetes/kubernetes/blob/bdc948d4c949bc498f8e599797e24a1726a99d69/CHANGELOG-1.14.md#v1140-alpha1). Not sure if I came across as suggesting a new shorthand or if I misread your comment, but for clarity's sake there it is.

**Comment by ORESoftware** [comment]
the problem is that the nixos-rebuild command is not available on non-NixOS linux machines.
However, the underlying issue is that if we run `nix-build` we get this error:
